packages/core/src/libraries/protected-app.ts

```diff
@@ -61,6 +61,46 @@ const getDefaultDomain = async () => {
   return domain;
 };
 
+/**
+ * Build application data for protected app
+ * check if subdomain is valid
+ * generate host based on subdomain
+ * generate default protectedAppMetadata based on host and origin
+ * generate redirectUris and postLogoutRedirectUris based on host
+ */
+const buildProtectedAppData = async ({
+  subDomain,
+  origin,
+}: {
+  subDomain: string;
+  origin: string;
+}): Promise<Pick<Application, 'protectedAppMetadata' | 'oidcClientMetadata'>> => {
+  assertThat(
+    isValidSubdomain(subDomain),
+    new RequestError({
+      code: 'application.invalid_subdomain',
+      status: 422,
+    })
+  );
+
+  // Skip for integration test, use empty value instead
+  const { domain } = EnvSet.values.isIntegrationTest ? { domain: '' } : await getProviderConfig();
```

Comment:
Feels like there might be multiple places calling getProviderConfig. should we place the `isIntegrationTest`  check within this method?

Reply:
Wait until necessary?

Reply:
fine to me if this is the only place